src/tests/histgen.cc

```diff
@@ -0,0 +1,114 @@
+// Copyright 2019 SMF Authors
+//
+
+#include <chrono>
+#include <iostream>
+#include <random>
+
+#include <seastar/core/app-template.hh>
+#include <seastar/core/distributed.hh>
+#include <seastar/core/metrics_registration.hh>
+#include <seastar/core/prometheus.hh>
+#include <seastar/http/httpd.hh>
+
+#include "smf/histogram.h"
+
+struct histgen_server_args {
+  seastar::sstring ip = "";
+  uint16_t port = 33140;
+  uint64_t min_val;
+  uint64_t max_val;
+  uint32_t num_samples;
+};
+
+class histgen_server {
+ public:
+  histgen_server(histgen_server_args args)
+    : args(args), hist(smf::histogram::make_lw_shared()),
+      http_server("histgen server") {
+    namespace sm = seastar::metrics;
+    metrics.add_group(
+      "histgen_server",
+      {
+        sm::make_histogram(
+          "synthetic_histogram", sm::description("Synthetic histogram data"),
+          [this] { return hist->get_seastar_metrics_histogram(); }),
+      });
+  }
+
+  ~histgen_server() {}
+
+  seastar::future<>
+  start() {
```

Comment:
this method confuses the non expert eye since `start` is what calls the constructor on the `distributed` seastar type. 

We should call this `do()` or smth like that.